src/systems/advanced_lift_drag/AdvancedLiftDrag.cc

```diff
@@ -0,0 +1,840 @@
+      half_rho_vel) + (this->Cenr * (yr*span/2) * half_rho_vel)) *
+        (this->area * span * body_z_axis);
+
+  // Compute moment (torque)
+  gz::math::Vector3d moment = pm+rm+ym;
+
+  // Compute force about cg in inertial frame
+  gz::math::Vector3d force = lift + drag + sideforce;
+  gz::math::Vector3d torque = moment;
+
+  // Correct for nan or inf
+  force.Correct();
+  this->cp.Correct();
+  torque.Correct();
+
+  // \todo(addisu) Create a convenient API for applying forces at offset
+  // positions
+  const auto totalTorque = torque + cpWorld.Cross(force);
+  Link link(this->linkEntity);
+  link.AddWorldWrench(_ecm, force, totalTorque);
+}
+
+//////////////////////////////////////////////////
+void AdvancedLiftDrag::Configure(const Entity &_entity,
+                         const std::shared_ptr<const sdf::Element> &_sdf,
+                         EntityComponentManager &_ecm, EventManager &)
+{
+  this->dataPtr->model = Model(_entity);
+  if (!this->dataPtr->model.Valid(_ecm))
+  {
+    gzerr << "Advanced LiftDrag system should be attached to a model entity."
+           << "Failed to initialize." << std::endl;
+    return;
+  }
+  this->dataPtr->sdfConfig = _sdf->Clone();
+}
+
+//////////////////////////////////////////////////
+void AdvancedLiftDrag::PreUpdate(const UpdateInfo &_info,
+  EntityComponentManager &_ecm)
+{
+  GZ_PROFILE("AdvancedLiftDrag::PreUpdate");
+
+  // \TODO(anyone) Support rewind
+  if (_info.dt < std::chrono::steady_clock::duration::zero())
+  {
+    gzwarn << "Detected jump back in time ["
+        << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
+        << "s]. System may not work properly." << std::endl;
+  }
+
+  if (!this->dataPtr->initialized)
+  {
+    // We call Load here instead of Configure because we can't be guaranteed
+    // that all entities have been created when Configure is called
+    this->dataPtr->Load(_ecm, this->dataPtr->sdfConfig);
+    this->dataPtr->initialized = true;
+
+    if (this->dataPtr->validConfig)
+    {
+      Link link(this->dataPtr->linkEntity);
+      link.EnableVelocityChecks(_ecm, true);
+
+      for(int i = 0; i < this->dataPtr->num_ctrl_surfaces; i++){
+
+        if ((this->dataPtr->controlJoints[i]!= kNullEntity) &&
+          !_ecm.Component<components::JointPosition>(this->dataPtr->
+            controlJoints[i]))
+          {
+            _ecm.CreateComponent(this->dataPtr->controlJoints[i],
+                components::JointPosition());
+          }
+      }
+    }
+  }
+
+  if (_info.paused)
+    return;
+
+  // This is not an "else" because "initialized" can be set in the if block
+  // above
+  if (this->dataPtr->initialized && this->dataPtr->validConfig)
+  {
+
+    this->dataPtr->Update(_ecm);
+  }
+}
+
+GZ_ADD_PLUGIN(AdvancedLiftDrag,
+                    System,
+                    AdvancedLiftDrag::ISystemConfigure,
+                    AdvancedLiftDrag::ISystemPreUpdate)
```

Comment:
```suggestion
                    AdvancedLiftDrag::ISystemPreUpdate)
                    
      GZ_ADD_PLUGIN_ALIAS(AdvancedLiftDrag,
                    "gz::sim::systems::AdvancedLiftDrag")
```

You still need the alias to be kept. Otherwise I get the following error:
```
[Err] [SystemLoader.cc:185] Failed to load system plugin: (Reason: library does not contain requested plugin)
- Requested plugin name: [gz::sim::systems::AdvancedLiftDrag]
- Requested library name: [gz-sim-advanced-lift-drag-system]
- Resolved library path: [/home/arjoc/ws/gz/garden/install/lib/gz-sim-7/plugins/libgz-sim-advanced-lift-drag-system.so]
- Detected Plugins:
  - gz::sim::v7::systems::AdvancedLiftDrag

```